app/controllers/memberships_controller.rb

```diff
@@ -95,4 +97,8 @@ def validate_can_manage_team
       raise ActionController::RoutingError, "Not Found"
     end
   end
+
+  def validate_has_two_confirmed_admin_users
+    raise ActionController::RoutingError, "Not Found" if @membership.cannot_change_permissions?
```

Comment:
I'm not sure this works. If you have 2 admins and remove one, this validation runs before the admin is removed and therefore passes. Also, suppose, you have one administrator user and one view_only user and you want to change the view_only user to administrator this validation will I think not allow you to do this.

Usually, with this kind of thing, the model is changed in memory and then validated before persisting to the database. Maybe we should have think on how to do this here because it may end up being quite difficult to do.